### PR TITLE
Initial Skeleton

### PR DESCRIPTION
**Pull request by DeeMac98**
Added main skeleton layout of classes, servlets and interfaces required. 
Some new classes may be added to represent multi-return values used in some of the Go functions.
Updated pom.xml with servlet dependency.
No use cases fulfilled yet. Once some Go specific syntax is converted to Java, I will be able to add functionality to the skeleton.

### CODE REVIEW

**Review by markpatton — CHANGES_REQUESTED** [review]
The Java package structure should start with org.eclipse.pass.policy

Update the README for the Java project it doesn't have to be complete, but make it accurate for the proposed design. That design should include the HTTP request and responses as well as status codes, headers, and parameters.

InternalPassClient and related clases are not needed. You will be using the existing Java PASS client. lesshttps://github.com/eclipse-pass/pass-doi-service/blob/main/src/main/java/org/dataconservancy/pass/doi/service/PassDoiServlet.java for an example of how to use it.

There are a bunch of Servlets defined to handle the endpoints /BaseURIs, /Client, /PolicyEndpoint, /PolicyService, and /RepositoriesEndpoint. It's not clear to me what all of those are doing. The README should explain what each endpoint does. I would think about having a higher level service handle the business logic and have the servlets manage the web request and dispatch it to the service.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+name: pass-policy-service Continuous Integration
```
Comment (markpatton) [inline .github/workflows/ci.yml]: This sets up Go and tries to use it to run tests. Take a look at a Java repository like pass-doi-service and copy the ci.yml from there.

File: src/main/java/org/eclipse/pass/policy/services/README.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
+# Web API
```
Comment (markpatton) [inline src/main/java/org/eclipse/pass/policy/services/README.md]: Put this in the README.md at the root of the repository. It should not live in a src directory.

File: src/main/java/org/eclipse/pass/policy/interfaces/BaseURIReplacer.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package org.eclipse.pass.policy.interfaces;
+
+/**
+ * Represents BaseURIReplacer interface
+ * Contains methods for replacing private with public URIs & vice versa
+ * @author David McIntyre
+ */
+public interface BaseURIReplacer {
```
Comment (markpatton) [inline src/main/java/org/eclipse/pass/policy/interfaces/BaseURIReplacer.java]: Is there a need to do this sort of replacement if you are using PassClient?

File: src/main/java/org/eclipse/pass/policy/rules/README.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
+# Policy rules DSL
```
Comment (markpatton) [inline src/main/java/org/eclipse/pass/policy/rules/README.md]: I would not have this in a src directory. You could have a separate directory for documentation like doc/

File: src/main/java/org/eclipse/pass/policy/services/PolicyEndpoint.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Policy Servlet
+ * Handles requests & responses relating to policies
+ * @author David McIntyre
+ */
+@WebServlet("/PolicyEndpoint")
```
Comment (markpatton) [inline src/main/java/org/eclipse/pass/policy/services/PolicyEndpoint.java]: Use the same endpoint as before, /policies.

File: src/main/java/org/eclipse/pass/policy/services/PolicyEndpoint.java [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+public class PolicyEndpoint extends HttpServlet {
```
Comment (markpatton) [inline src/main/java/org/eclipse/pass/policy/services/PolicyEndpoint.java]: Call this PolicyServlet

File: src/main/java/org/eclipse/pass/policy/services/RepositoriesEndpoint.java [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Repositories Servlet
+ * Handles requests & responses relating to repositories
+ * @author David McIntyre
+ */
+@WebServlet("/RepositoriesEndpoint")
```
Comment (markpatton) [inline src/main/java/org/eclipse/pass/policy/services/RepositoriesEndpoint.java]: Use the same endpointname as before /repositories. Also I'd change the name to RepositoriesServlet.

File: src/main/webapp/WEB-INF/web.xml [hunk 8]
```diff
@@ -0,0 +1,7 @@
+<!DOCTYPE web-app PUBLIC
```
Comment (markpatton) [inline src/main/webapp/WEB-INF/web.xml]: Actually should not be needed with newer application containers, the annotations should be enough. But no harm in having.

File: src/main/webapp/index.jsp [hunk 9]
```diff
@@ -0,0 +1,6 @@
+<!DOCTYPE html>
```
Comment (markpatton) [inline src/main/webapp/index.jsp]: Should not have this. This won't be doing anything else besides answering the endpoints.

**Review by markpatton — CHANGES_REQUESTED** [review]
This is getting closer. Take a look at the comments and see if you have questions.

**Comment by DeeMac98** [comment]
Updated issue: [Issue Tracker](https://github.com/eclipse-pass/main/issues/340)

**Review by markpatton — APPROVED** [review]

**Comment by markpatton** [comment]
Looks good!

**Pull request merged**